.github/workflows/test.yml

```diff
@@ -76,6 +76,27 @@ jobs:
     - name: Test EfficientNet
       run: curl https://media.istockphoto.com/photos/hen-picture-id831791190 | ./recognize | grep hen
 
+  testcpuwindows:
+    name: CPU on Windows
+    runs-on: windows-latest
+    timeout-minutes: 20
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v3
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.8
+    - name: Cache pip
+      uses: actions/cache@v3
+      with:
+        path: '~\AppData\Local\pip\cache'
+        key: testing
+    - name: Install Dependencies
+      run: pip install -e '.[testing]' --extra-index-url https://download.pytorch.org/whl/cpu
+    - name: Run pytest
+      run: python -m pytest -n=auto test/ -k 'test'
+
```

Comment:
Do we need the windows cpu test?

Reply:
This is the current CI behavior. Maybe the intention was to only have CLANG for windows, but it was configured incorrectly. I added these to not drop overall test coverage, and it can be a separate PR to fix if CLANG windows is what we want. The main goal of this PR is to fix clang tests that was dropped in a speed optimization PR.

Reply:
I think the original intention was to only have the windows clang test

Reply:
Are you suggesting removing the windows CPU coverage? I can do that too. My first commit did not have this section.  @dc-dc-dc suggested that it's dropping windows coverage so I added it back as a standalone test.

The first commit #1011 was configured incorrectly, so we probably never have windows clang in CI. It just showed that even though the correct windows + clang might be easy to implement, it's very easy to miss and we need to verify carefully. That's why I don't want to bundle correcting windows + clang with bringing back ubuntu + clang in the same PR. I think fixing CI missing tests is more important / urgent than adding new CI functionalities.

Reply:
Just use this pr to fix windows+clang, because that's what's most important right now.

Reply:
I think @dc-dc-dc meant to keep the clang test on windows, not keep the windows+cpu test, tho correct me if im wrong

Reply:
I disagree. The most important thing right now is adding back ubuntu clang test, it was the clang test that worked, and got dropped accidentally.

I don't mind adding/removing the windows CPU part. Let me know if you prefer it removed and I will update the PR.

There was no working windows clang tests to start with, so no "keep the clang tests on windows". Adding that is adding a new CI feature, not a fix, and I don't understand prioritizing new feature over a fix.

Reply:
Oh, that's my bad, didn't read the original comment carefully enough. I thought that the the windows+clang tests worked before the pr.

We definitely do need the windows+clang tests tho, because that exact combination has broken before.

I'll merge adding ubuntu+clang back for now because we do need a working clang test, then we can prioritize switching it to windows+clang. Unless you want to just fix windows+clang in this pr.

Reply:
Let's merge this one first.

We can have a follow up PR that replace the windows CPU and ubuntu clang with a windows + clang test. I can look into it later, unless you already know how to do it? I imagine it's not very difficult but need to be extra careful to prevent regression like this from happening again.